build/commands/scripts/mcp-setup.js

```diff
@@ -0,0 +1,415 @@
+  if (fs.existsSync(claudeUserDir) || process.env.CLAUDE_CODE_API_KEY) {
+    ides.push('claude');
+  }
+  
+  // Check for Cursor (look for .cursor directory or in PATH)
+  const cursorUserDir = path.join(require('os').homedir(), '.cursor');
+  if (fs.existsSync(cursorUserDir)) {
+    ides.push('cursor');
+  }
+  
+  // Try to detect from running processes or common install locations
+  try {
+    if (isWindows) {
+      // Check common Windows install paths
+      const cursorPaths = [
+        path.join(require('os').homedir(), 'AppData', 'Local', 'Programs', 'cursor'),
+        'C:\\Users\\' + require('os').userInfo().username + '\\AppData\\Local\\Programs\\cursor'
+      ];
+      if (cursorPaths.some(p => fs.existsSync(p))) {
+        if (!ides.includes('cursor')) ides.push('cursor');
+      }
+    } else if (isMacOS) {
+      // Check macOS application directories
+      const cursorPaths = [
+        '/Applications/Cursor.app',
+        path.join(require('os').homedir(), 'Applications', 'Cursor.app')
+      ];
+      if (cursorPaths.some(p => fs.existsSync(p))) {
+        if (!ides.includes('cursor')) ides.push('cursor');
+      }
+      
+      // Also check if cursor is in PATH on macOS
+      try {
+        execSync('which cursor', { stdio: 'pipe' });
+        if (!ides.includes('cursor')) ides.push('cursor');
+      } catch (e) {
+        // cursor not in PATH
+      }
+    } else if (isLinux) {
+      // Check Linux application directories and PATH
+      const cursorPaths = [
+        '/usr/bin/cursor',
+        '/usr/local/bin/cursor',
+        path.join(require('os').homedir(), '.local', 'bin', 'cursor'),
+        '/opt/cursor/cursor',
+        '/snap/bin/cursor'
+      ];
+      if (cursorPaths.some(p => fs.existsSync(p))) {
+        if (!ides.includes('cursor')) ides.push('cursor');
+      }
+      
+      // Also check if cursor is in PATH on Linux
+      try {
+        execSync('which cursor', { stdio: 'pipe' });
+        if (!ides.includes('cursor')) ides.push('cursor');
+      } catch (e) {
+        // cursor not in PATH
+      }
+    }
+  } catch (e) {
+    // Ignore detection errors
+  }
+  
+  return ides;
+}
+
+function createMCPConfig() {
+  log('Creating MCP server configuration...');
+  
+  // Determine paths based on execution context and platform
+  // Support running from brave-browser or brave-browser/src/brave
+  const currentDir = process.cwd();
+  let projectRoot, braveCorePath, chromiumSrcPath, pythonPath;
+  
+  if (currentDir.endsWith('src/brave') || currentDir.includes('/src/brave')) {
+    // Running from src/brave directory
+    projectRoot = path.resolve(currentDir, '../..');
+    chromiumSrcPath = path.resolve(currentDir, '..');
+  } else {
+    // Running from brave-browser root
+    projectRoot = currentDir;
+    chromiumSrcPath = path.join(currentDir, 'src');
+  }
+  
+  // Brave-core is typically a sibling to brave-browser
+  braveCorePath = path.resolve(projectRoot, '..', 'brave-core');
```

Comment:
nit: I'm not certain this assumption holds true. It would be good to get the @brave/build-commands-reviewers team to weigh in if I should take this or an alternative path.